mockgcp/mockbackupdr/testdata/managementserver/crud/_http.log

```diff
@@ -112,7 +112,7 @@ Content-Type: application/json
   "address": "8.8.8.8",
   "addressType": "INTERNAL",
   "name": "google-managed-services-${uniqueId}",
-  "network": "projects/${projectId}/global/networks/test-network-${uniqueId}",
+  "network": "https://compute.googleapis.com/compute/v1/projects/${projectId}/global/networks/test-network-${uniqueId}",
```

Comment:
I think removing the hack will cause the HTTP request to fail because the "network" field is incorrectly formatted?

Reply:
I'm not sure why gcloud behaves differently. When testing locally (e.g., in my terminal), the HTTP request sent by the gcloud command is correct, with `"network": "projects/${projectId}/global/networks/test-network-${uniqueId}"`. However, in our `mockgcptests/e2e_test`, gcloud sends an incorrect HTTP request (after proxied) where `"network"` appears as `"https://compute.googleapis.com/compute/v1/projects/${projectId}/global/networks/test-network-${uniqueId}"`. This format is rejected by the GCP API as invalid.

Reply:
Ah so I think this is actually what I'm trying to fix then!  There are two problems with api overrides: 1) not all services implement it, so we need a CONNECT proxy.  But also and more oddly (2) I noticed something similar where gcloud was constructing a URL incorrectly ... I think it was also a compute URL.  There is some fancy logic happening in gcloud to convert URLs (it appears), and I think that api_overrides changes that logic, which isn't what we want in a mock.

The easy fix therefore is to switch to the CONNECT proxy, which (hopefully) should not change gcloud behaviour because it is at the transport layer.  It is much more complicated to implement (and my initial implementation leaves much to be desired), but once we have it it seems more reliable!